account_payment_promissory_note/wizard/account_register_payments.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2018 Sergio Teruel <sergio.teruel@tecnativa.com>
+# Copyright 2018 Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class AccountRegisterPayments(models.TransientModel):
+    _name = "account.payment.register"
+    _inherit = ["account.payment.register", "account.promissory.note.mixin"]
+
+    def _create_payments(self):
+        payments = super()._create_payments()
+        for payment in payments:
+            if not self.date_due:
+                invoices = payment.reconciled_invoice_ids
+                max_date = max(invoices.mapped("invoice_date_due"))
+                payment.write(
```

Comment:
Use the update method or field assign directly to take advantage of improvements in the framework
```suggestion
                payment.update(
```